### DISCUSSION TITLE
Pulling from private repo

### DISCUSSION THREAD

**Discussion by AnimusXCASH** [gitpython-developers/GitPython#1146] [Q&A] [answered]
Hi, is there are possibility to pull from private repo, and where to provide username and password details?

Thank you for answer and time in advance

**Comment by Byron** [comment, marked as answer]
I think you will find what you need [in this part of the docs](https://gitpython.readthedocs.io/en/latest/tutorial.html#handling-remotes) - just search for 'ssh' to learn how to specify custom ssh keys.

Alternatively, and when pulling from github, you can use [HTTPS based authentication](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/https-cloning-errors#provide-an-access-token).

Please feel free to comment on the closed issue.
